spritzle/main.py

```diff
@@ -40,26 +41,45 @@
 from spritzle.logger import setup_logger
 
 
-async def debug_middleware(app, handler):
-    async def middleware(request):
-        if request.content_type in ('application/x-www-form-urlencoded',
-                                    'multipart/form-data'):
-            body = await request.post()
-        else:
-            body = await request.text()
-        log = app['spritzle.log']
-        log.debug('*'*20 + 'REQUEST' + '*'*20)
-        log.debug(f'URL: {request.rel_url}')
-        log.debug(f'METHOD: {request.method}')
-        log.debug(f'HEADERS: {request.headers}')
-        log.debug(f'BODY: {body}')
-        log.debug('*'*47)
-        return await handler(request)
-    return middleware
-
-app = aiohttp.web.Application(
-    middlewares=[debug_middleware,
-                 spritzle.resource.auth.auth_middleware])
+@aiohttp.web.middleware
+async def debug_middleware(request, handler):
+    if request.content_type in ('application/x-www-form-urlencoded',
+                                'multipart/form-data'):
+        body = await request.post()
+    else:
+        body = await request.text()
+    log = request.app['spritzle.log']
+    log.debug('*' * 20 + 'REQUEST' + '*' * 20)
+    log.debug(f'URL: {request.rel_url}')
+    log.debug(f'METHOD: {request.method}')
+    log.debug(f'HEADERS: {request.headers}')
+    log.debug(f'BODY: {body}')
+    log.debug('*' * 47)
+    return await handler(request)
+
+
+@aiohttp.web.middleware
+async def error_middleware(request, handler):
+    try:
+        response = await handler(request)
+    except aiohttp.web.HTTPException as ex:
+        response = ex
+    except Exception:
+        # Unhandled exception, this is a bug in Spritzle
+        tb = ''.join(traceback.format_exception(*sys.exc_info()))
+        response = aiohttp.web.Response(status=500, reason='Spritzle Bug',
+                                        text=tb)
+    if 200 <= response.status < 300:
+        return response
+    return aiohttp.web.json_response({
+        'error': True,
```

Comment:
How is this key used? Seems like it's not necessary since we're still using HTTP status codes to indicate errors.

Reply:
True. I was just thinking the client could check for errors that way. You are right though, it can use the status codes.

Reply:
The one way I've seen it done is return an object with an 'error' key that contains a nested dictionary, that way you can do a simple 'if error in response'. I don't really think we should go down that path though.

Reply:
Yeah, the `if 'error' in response` was what I was thinking of by adding it. I'm good with keeping it in just status codes though.